lib/language.js

```diff
@@ -0,0 +1,118 @@
+// Load modules
+
+var Boom = require('boom');
+var Hoek = require('hoek');
+
+
+// Declare internals
+
+var internals = {};
+
+
+// https://tools.ietf.org/html/rfc7231#section-5.3.5
+// Accept-Language: da, en-gb;q=0.8, en;q=0.7
+
+
+exports.language = function (header, preferences) {
+
+    Hoek.assert(!preferences || Array.isArray(preferences), 'Preferences must be an array');
+    var languages = exports.languages(header);
+
+    if (languages.length === 0) {
+        languages.push('');
+    }
+
+    // No preferences.  Take the first charset.
+
+    if (!preferences || preferences.length === 0) {
+        return languages[0];
+    }
+
+    // If languages includes * return first preference
+
+    if (languages.indexOf('*') !== -1) {
+        return preferences[0];
+    }
+
+    // Try to find the first match in the array of preferences
+
+    preferences = preferences.map(function (str) {
+
+        return str.toLowerCase();
+    });
+
+    for (var i = 0, il = languages.length; i < il; ++i) {
+        if (preferences.indexOf(languages[i].toLowerCase()) !== -1) {
```

Comment:
isn't `languages` something you have control over? why not just store the data in lowercase so you don't have to waste the overhead of calling `.toLowerCase()` so often?